libdnf/advisory/advisory_set.cpp

```diff
@@ -137,5 +138,22 @@ BaseWeakPtr AdvisorySet::get_base() const {
     return p_impl->base;
 }
 
+std::vector<AdvisoryPackage> AdvisorySet::get_advisory_packages_sorted_by_id(bool only_applicable) const {
+    std::vector<AdvisoryPackage> out;
+    for (Id candidate_id : *p_impl) {
+        Advisory advisory2 = Advisory(p_impl->base, AdvisoryId(candidate_id));
+        auto collections = advisory2.get_collections();
+        for (auto & collection : collections) {
+            if (only_applicable && !collection.is_applicable()) {
+                continue;
+            }
+            collection.get_packages(out);
+        }
+    }
+
+    std::sort(out.begin(), out.end(), AdvisoryPackage::Impl::nevra_compare_lower_id);
```

Comment:
I think the `AdvisoryPackage::Impl::nevra_compare_lower_id` and friends could be pulled out of that class so that you don't need a friend and semantics-wise they are the same as for sack packages, right? Maybe a shared implementation? Templated, probably.

Treat this as a remark for now, no need to solve it here.